src/main/java/com/aws/greengrass/clientdevices/auth/iot/CertificateRegistry.java

```diff
@@ -102,19 +103,22 @@ public void updateCertificate(Certificate certificate) {
     }
 
     /**
-     * Returns the PEMs that are stored for all the certificate ids that are being tracked.
+     * Deletes a certificate from the repository.
+     *
+     * @param certificate certificate to remove
      */
-    public String[] getAllCertificateIds() {
-        return runtimeConfiguration.getAllCertificateIdsV1();
+    public void deleteCertificate(Certificate certificate) {
+        deleteCertificate(certificate.getCertificateId());
+        pemStore.removePem(certificate.getCertificateId());
     }
 
     /**
-     * Deletes a certificate from the repository.
+     * Deletes a certificate from the repository provided its id.
      *
-     * @param certificate certificate to remove
+     * @param certificateId - certificateId of the certificate to remove
      */
-    public void deleteCertificate(Certificate certificate) {
-        runtimeConfiguration.removeCertificateV1(certificate.getCertificateId());
+    public void deleteCertificate(String certificateId) {
+        runtimeConfiguration.removeCertificateV1(certificateId);
     }
```

Comment:
make this function `private` ? so that exposed `public void deleteCertificate(Certificate certificate) {}` will always remove registry entry as well as PEM.